app/src/main/java/org/wikipedia/feed/onthisday/OnThisDayCardView.java

```diff
@@ -133,42 +99,99 @@ private void header(@NonNull OnThisDayCard card) {
-        getContext().startActivity(OnThisDayActivity.newIntent(getContext(), age, getCard().wikiSite(),
-                ON_THIS_DAY_CARD_BODY), options.toBundle());
+        ActivityOptionsCompat options = ActivityOptionsCompat.makeSceneTransitionAnimation((Activity) getContext(),
+                headerView.getTitleView(), getContext().getString(R.string.transition_on_this_day));
+        getContext().startActivity(OnThisDayActivity.newIntent(getContext(), age, isYearClicked ? getCard().year() : -1,
+                getCard().wikiSite(), isYearClicked ? ON_THIS_DAY_CARD_YEAR : ON_THIS_DAY_CARD_BODY), options.toBundle());
     }
 
-    @OnClick({R.id.more_events_layout}) void onMoreFooterClick() {
-        funnel.cardClicked(CardType.ON_THIS_DAY, getCard().wikiSite().languageCode());
-        ActivityOptionsCompat options = ActivityOptionsCompat.
-                makeSceneTransitionAnimation((Activity) getContext(), dayTextView, getContext().getString(R.string.transition_on_this_day));
-        getContext().startActivity(OnThisDayActivity.newIntent(getContext(), age, getCard().wikiSite(),
-                ON_THIS_DAY_CARD_FOOTER), options.toBundle());
-    }
-
-    private void setPagesRecycler(OnThisDayCard card) {
+    private void updateOtdEventUI(OnThisDayCard card) {
+        pagesViewPager.setVisibility(GONE);
+        PageSummary chosenPage = null;
         if (card.pages() != null) {
-            RecyclerAdapter recyclerAdapter = new RecyclerAdapter(((AppCompatActivity) getContext()).getSupportFragmentManager(), card.pages(), card.wikiSite(), true);
-            pagesRecycler.setAdapter(recyclerAdapter);
+            otdEventView.setVisibility(VISIBLE);
+            for (PageSummary pageSummary : card.pages()) {
+                chosenPage = pageSummary;
+                if (pageSummary.getThumbnailUrl() != null) {
+                    break;
+                }
+            }
+            PageSummary finalChosenPage = chosenPage;
+
+            if (chosenPage != null) {
+                if (chosenPage.getThumbnailUrl() == null) {
+                    otdEventImage.setVisibility(GONE);
+                } else {
+                    otdEventImage.setVisibility(VISIBLE);
+                    otdEventImage.loadImage(Uri.parse(chosenPage.getThumbnailUrl()));
+                }
+                otdEventDescription.setText(chosenPage.getDescription());
+                otdEventDescription.setVisibility(TextUtils.isEmpty(chosenPage.getDescription()) ? View.GONE : View.VISIBLE);
+                otdEventTitle.setMaxLines(TextUtils.isEmpty(chosenPage.getDescription()) ? 2 : 1);
+                otdEventTitle.setText(StringUtil.fromHtml(chosenPage.getDisplayTitle()));
+                otdEventView.setOnClickListener(view -> {
+                    PageTitle pageTitle = finalChosenPage.getPageTitle(card.wikiSite());
+                    HistoryEntry entry = new HistoryEntry(pageTitle, HistoryEntry.SOURCE_ON_THIS_DAY_CARD);
```

Comment:
Since we will use the same entry here and below, would it be better if we create a method of it? (and maybe reduce the `final` variable?